lazyflow/operators/opCompressedUserLabelArray.py

```diff
@@ -247,12 +262,24 @@ def _executeProjection2D(self, roi, destination):
             # Compute slicing within the deep array and slicing within this block
             deep_relative_intersection = numpy.subtract(intersecting_roi, input_roi.start)
             block_relative_intersection = numpy.subtract(intersecting_roi, block_start)
-                        
-            deep_data = self._getBlockDataset( entire_block_roi )[roiToSlice(*block_relative_intersection)]
+            block_relative_intersection_slicing = roiToSlice(*block_relative_intersection)
+
+            block = self._getBlockDataset( entire_block_roi )
+            deep_data = None
+            if self.Output.meta.has_mask:
+                deep_data = numpy.ma.masked_array(
+                    block["data"][block_relative_intersection_slicing],
+                    mask=block["mask"][block_relative_intersection_slicing],
+                    fill_value=block["fill_value"][()],
+                    shrink=False
+                )
+            else:
+                deep_data = block[block_relative_intersection_slicing]
+
+            # make binary and convert to float (must copy)
+            deep_data_float = deep_data.astype(numpy.float32)
+            deep_data_float[deep_data_float.nonzero()] = 1
 
-            # make binary and convert to float
-            deep_data_float = numpy.where( deep_data, numpy.float32(1.0), numpy.float32(0.0) )
-            
             # multiply by slice-index
```

Comment:
Just as a follow-up, I looked into `numpy.where`. This appears to be implemented in C, which would make it a bit tricky to add a fix for `MaskedArray`s as they are implemented in Python. One way to address this, might be to add a Python shim function in place of `numpy.where`, which then calls the C implementation of `numpy.where` or `numpy.ma.where` as needed. This would probably be the easiest way to fix this behavior.